### PR TITLE
Cache dependencies

### PR DESCRIPTION
**Pull request by bsels**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -22,6 +30,16 @@ jobs:
...
           ghc-version: '8.10.3'
           cabal-version: '3.4'
+      - uses: actions/cache@v2
+        name: Cache cabal
+        with:
+          path: |
+            ~/.cabal
+            ~/.ghc
+            dist-newstyle
+          key: ${{ runner.os }}-cabalghc-20210330
```
Comment (niknetniko) [inline .github/workflows/ci.yml:40]: Why is this a hardcoded date?

Comment (bsels) [reply .github/workflows/ci.yml:40]: I'm not sure how we use a generic hash for the cabal dependencies.
So a hardcoded date may do the trick to get a unique hashkey.

Comment (niknetniko) [reply .github/workflows/ci.yml:40]: Reading online we probably need `cabal freeze`, but unsure how this works with global v1 installation commands, so perhaps we can do it later if/when we update to v2 commands.

**Review by niknetniko — APPROVED** [review]
Definitely an improvement.

For the future we might look into upgrading to GHC 9; this is already included in the GA image, so the GHC installation step should become shorter.

**Pull request merged**